### ISSUE TITLE
GitHub Branch Source page - continuous-integration/jenkins/branch build never starts for fork PRs

### ISSUE THREAD

**Issue by danielolsonah** [jenkins-infra/plugin-site#496]
Problem with the [GitHub Branch Source](https://plugins.jenkins.io/github-branch-source) page, [source file](https://github.com/jenkins-infra/plugin-site/tree/master/src/templates/plugin.jsx)

For a non-fork PR (both branches are in original repo) there are 2 status checks, one for continuous-integration/jenkins/branch and one for continuous-integration/jenkins/pr-merge.  Both of them start corresponding builds on Jenkins, and when they finish the result is sent back to github.  

**Unexpected Behavior**
When I create a fork, then create a PR from one of my fork's branches to one of the original repo's branches, the continuous-integration/jenkins/pr-merge and continuous-integration/jenkins/branch status checks automatically show up in the PR conversation page as expected, but the continuous-integration/jenkins/branch never starts a build anywhere.  So it is stuck in this "Expected" state indefinitely, as shown in screenshot below.

tl;dr Jenkins discovers the PR from fork, github expects a branch build status check, Jenkins doesn't respond and githubb is stuck "Expecting" a result forever

## Screenshots 

![image](https://user-images.githubusercontent.com/37480879/100285998-d2fd2400-2f26-11eb-897c-4cd7f4f7e631.png)

## Possible Solution 

I'm afraid I do not have any intelligent ideas for solutions

N/A

**Comment by halkeye** [comment]
Hi, 

Thank you very much for the report, but the issue tracker for the plugins.jenkins.io site is used to track issues with the plugin site itself. Your request seems to be a request for help, not a report of a functionality issue or improvement.

Please use the [mailing lists](https://www.jenkins.io/mailing-lists/) and the [chat channels](https://www.jenkins.io/chat/) for questions or even create a bug on the [issues site](https://issues.jenkins.io)
